protocols/virtualfund/virtualfund.go

```diff
@@ -137,7 +137,14 @@ type Objective struct {
 
 // NewObjective creates a new virtual funding objective from a given request.
 func NewObjective(request ObjectiveRequest, preApprove bool, myAddress types.Address, getTwoPartyConsensusLedger GetTwoPartyConsensusLedgerFunction) (Objective, error) {
-	rightCC, ok := getTwoPartyConsensusLedger(request.Intermediaries[0])
+	var rightCC *consensus_channel.ConsensusChannel
+	ok := false
+
+	if len(request.Intermediaries) > 0 {
+		rightCC, ok = getTwoPartyConsensusLedger(request.Intermediaries[0])
```

Comment:
It might make sense to separate out the support for 0-hop channels, since that is quite a major conceptual leap (even if it is one that seems to make good sense)?

Or, if we want this to go in now we can make some updates to documentation somewhere and point to them here?

Reply:
I'm in favour of putting it in - I am still in progress with the docs site review and should be able to suggest the appropriate changes quickly.